ssha256/ssha256_test.go

```diff
@@ -35,7 +35,7 @@ func TestGenerate4(t *testing.T) {
 		return
 	}
 
-	if res, err = ssha256.Validate(pass, hash); err != nil && res != false {
+	if res, err = ssha256.Validate(pass, hash); err != nil || res != true {
```

Comment:
What exactly is the reason for this inverse logic? Was that a result of debugging you did while finding the fix?

Reply:
Hah, nevermind :) I found the answer myself. It turned out I had an inconsistency in the testing between different hash algorithms.

Reply:
Yup, it was just a correction because it seemed like it was a copy-paste from the "invalid test" rather than the "valid test" and we're expecting a `true`, so we want to report a `false` or an `err` and not both an `err` and a `true`.